system/Bootstrap.cfc

```diff
@@ -476,9 +477,10 @@ component serializable="false" accessors="true" {
 	boolean function onRequestStart( required targetPage ) output=true{
 		// Global flag to denote if we are in mid reinit or not.
 		cfparam( name = "application.fwReinit", default = false );
+		cfparam( name = "request.isReinitRequestor", default = false );
 
 		// Fail fast so users coming in during a reinit just get a please try again message.
-		if ( application.fwReinit ) {
+		if ( application.fwReinit && !request.isReinitRequestor ) {
```

Comment:
I am not sure how the `request.isReinitRequestor` be of use, since it only is attached to a single visitor request.  And that request is not threaded.  I am trying to see why this should be here.

Reply:
The problem is that `onRequestStart` gets called twice in a reinit, which allows a collision if a request comes in during the reinit- which then shows the "Fail Fast" message to the reinit requestor ( while the reinit continues in the logs in the background ). Effectively, the requestor gets trapped by its own reinit flag.

Reply:
I think we have to keep this, as insurance from the reinit requestor getting trapped in the fail fast